src/Platforms/SQLitePlatform.php

```diff
@@ -934,6 +934,11 @@ private function getForeignKeysInAlteredTable(TableDiff $diff): array
             $constraintName = $constraint->getName();
 
             if ($constraintName === '') {
+                Deprecation::trigger(
```

Comment:
Whether a foreign key constraint is dropped on SQLite or any other platform, it's expected to have a name: https://github.com/doctrine/dbal/blob/aa6b906fcf0cf09e3adde1c623131accd55de63b/src/Platforms/AbstractPlatform.php#L1372-L1374

The fact that `TableDiff::getDroppedForeignKeys()` can return unnamed constraints is essentially a bug.

I think we should move this deprecation to the `TableDiff` constructor. In 5.0.x, this deprecation will need to be promoted to an exception, and all callers of `TableDiff::getDroppedForeignKeys()` could make an assertion that the constraint being dropped does have a name before using it.